chemprop/nn/loss.py

```diff
@@ -75,6 +73,13 @@ def __call__(
     def forward(self, preds, targets, mask, w_s, w_t, lt_mask, gt_mask) -> Tensor:
         """Calculate a tensor of shape `b x t` containing the unreduced loss values."""
 
+    def extra_repr(self) -> str:
+        sig_params = inspect.signature(self.__class__).parameters
+        obj_params = self.__dict__.items()
+        items = [(k, v) for k, v in obj_params if (k in sig_params and v != sig_params[k].default)]
+
+        return ", ".join(f"{k}={repr(v)}" for k, v in items)
```

Comment:
Feel like it's not worth it to set up a generic implementation of the function. Rather, each class should define its own.

Reply:
I'm not sure how to write a class specific implementation. We will still need to call `inspect.signature` to get the defaults (so that we don't print the value if it is the default). We will also still need to use `self.__dict__.items()` to get the current values. 
I think the only benefit of a class specific versus general version is we wouldn't need to check `k in sig_params` because we know what the keys should be. 
Given the current generic implementation works and reduces copying code, I prefer not switching to class implementations.

Reply:
Ex:

```python
class SIDLoss(nn.Module):
    def extra_repr(self):
        if self.threshold is None:
            return f"threshold={self.threshold}" 

        return f"threshold={self.threshold:0.2f}" 
...
loss = SIDLoss(threshold=0.5)
print(loss)
# SIDLoss(threshold=0.5)
```

Finding some generic way to implmenet `extra_repr` is probably more trouble than it's worth is my point. Using the `inspect` module is a fairly brittle to dynamically generate the string, and I don't think it's that bad to actually just hard-code the behavior in subclasses that would benefit from it.